### PR TITLE
[GEOT-5302] remove support for EnvFunction lookup tables

### PR DESCRIPTION
**Pull request by fgdrf**
see https://osgeo-org.atlassian.net/browse/GEOT-5302 for details and comments.

Implemented option 1 and added two functions that provides remove support. In addition I updated the javadoc comment for setGlobalValue

### CODE REVIEW

File: modules/library/main/src/main/java/org/geotools/filter/function/EnvFunction.java [hunk 1]
```diff
@@ -90,6 +90,7 @@
  *
```
Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/filter/function/EnvFunction.java]: update header

**Comment by jodygarnett** [comment]
Do we have any documentation on how EnvFunctions work that should be updated as part of this pull request?

**Comment by fgdrf** [comment]
Update javadoc and added serveral SLD & code java examples how to use isNull and isNil. Added serveral test cases to verify behavior.
EnvFunction.setGlobalValue() now supports null values, changed internal Map from ConcurrentHashMap to a synchronized HashMap to be thread save for gets and puts.

File: modules/library/main/src/main/java/org/geotools/filter/function/EnvFunction.java [hunk 2]
```diff
@@ -20,12 +20,12 @@
 import static org.geotools.filter.capability.FunctionNameImpl.parameter;
```
Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/filter/function/EnvFunction.java]: Please update header

**Comment by jodygarnett** [comment]
Other then updating headers this one is good to go.

**Comment by bencaradocdavies** [comment]
@fgdrf I think this was manually merged by @jodygarnett on 14.x and master.

**Pull request closed without merging**